reviveme/api/v1/vote_controller.py

```diff
@@ -0,0 +1,94 @@
+from . import bp
+from flask import Response, request
+from marshmallow import Schema, fields
+
+from reviveme import db
+from reviveme.models import ThreadVote, CommentVote, Thread, Comment
+
+from sqlalchemy import select
+
+class UpVoteSchema(Schema): # TODO: move these schema into separate files
+    upvote = fields.Boolean(required=True)
+    user_id = fields.Int(required=True)
+    
+class DownVoteSchema(Schema):
+    downvote = fields.Boolean(required=True)
+    user_id = fields.Int(required=True)
+
+def handle_upvote(VoteClass, item_id):
+    data = UpVoteSchema().load(request.get_json())
+
+    vote = db.session.execute(
+        select(VoteClass).where(VoteClass.item_id == item_id, VoteClass.user_id == data['user_id'])
+    ).scalars().first()
+
+    if vote is None and data['upvote']: # if no vote and user wants to upvote
+        vote = VoteClass(user_id=data['user_id'], item_id=item_id, upvote=True)
+        db.session.add(vote)
+    elif vote.upvote and not data['upvote']: # if user upvoted and now wants to remove upvote
+        db.session.delete(vote)
+    elif not vote.upvote and data['upvote']: # if user downvoted and now wants to upvote
+        vote.upvote = True
+        db.session.add(vote)
```

Comment:
does this add a new entry into the table or update it?

Reply:
It updates it. To be honest, this line came from copilot LOL. I don't actually think the `db.session.add` is needed here, since the code works without it. I'll remove it.